packages/framework/fluid-static/src/fluidContainer.ts

```diff
@@ -10,7 +10,23 @@ import {
 	ICriticalContainerError,
 	ConnectionState,
 } from "@fluidframework/container-definitions";
-import type { IRootDataObject, LoadableObjectClass, LoadableObjectRecord } from "./types";
+import type { ContainerSchema, IRootDataObject, LoadableObjectClass } from "./types";
+
+/**
+ * Extract the type of 'initialObjects' from the given ContainerSchema type.
+ */
+export type InitialObjects<T extends ContainerSchema> = {
+	// Construct a LoadableObjectRecord type by enumerating the keys of
+	// 'ContainerSchema.initialObjects' and infering the value type of each key.
+	//
+	// The '? TChannel : never' is required because infer can only be used in
+	// a conditional 'extends' expression.
+	[K in keyof T["initialObjects"]]: T["initialObjects"][K] extends LoadableObjectClass<
+		infer TChannel
+	>
+		? TChannel
+		: never;
```

Comment:
I don't understand this too well but just pointing out how @CraigMacomber commented on subtle errors we've had using this pattern (I think?) so might be worth double checking it's correct.